src/js/menu/menu.jsx

```diff
@@ -69,24 +79,25 @@ var NestedMenuItem = React.createClass({
   },
 
   _onParentItemClick: function() {
-    this.setState({ open: !this.state.open });
+    if (!this.props.disabled) this.setState({ open: !this.state.open });
   },
 
   _onMenuItemClick: function(e, index, menuItem) {
-    this.setState({ open: false });
     if (this.props.onItemClick) this.props.onItemClick(e, index, menuItem);
+    this.setState({ open: false });
```

Comment:
If the menu item does not have the opportunity to process the event first, a click on a disabled element will close the menu and then the disabled menu item will prevent the event.